gateway/ln-gateway/src/lib.rs

```
@@ -251,249 +310,82 @@ impl Gatewayd {
                                 // Blocks until the connection to the lightning node breaks
                                 info!("Established HTLC stream");
 
-                                // Re-create gateway
-                                let gateway = Gateway::new(
-                                    ln_client.clone(),
-                                    client_builder.clone(),
-                                    self.fees.clone().unwrap_or(GatewayFee(DEFAULT_FEES)).0,
-                                    database.clone(),
-                                    self.api_addr.clone(),
-                                    clients.clone(),
-                                    scid_to_federation.clone(),
-                                    tg.clone(),
-                                    self.num_route_hints,
-                                )
-                                .await.expect("Failed to created Gateway");
-
-                                info!("Successfully created Gateway");
-
-                                let task_group = run_webserver(self.password.clone(), self.listen, gateway)
-                                    .await
-                                    .expect("Failed to start webserver");
-                                info!("Successfully started webserver");
-
-                                Gateway::handle_htlc_stream(stream, ln_client, handle.clone(), scid_to_federation.clone(), clients.clone()).await;
-                                warn!("HTLC Stream Lightning connection broken. Stopping webserver...");
-                                task_group.shutdown();
+                                if let Ok((lightning_public_key, lightning_alias)) =
```

`match` here and log the `Err(e)`. Avoid swallowing errors